Bump Pygeoprocessing requirement, and a couple other fixes

davemfish opened the pull request:
- The last PR that included `gdal.UseExceptions` needed to also require we use a version of pygeoprocessing that does the same, so that exceptions there are handled properly, such as in `get_gis_type`
- There was also a bug in the "units" representation of the CRS, which is fixed here.
- Another bug in the `Config` test mocking is fixed (see Slack thread for explanation)
- Fixes #50

phargogh reviewed (approved):
Makes sense to me, thanks @davemfish !

davemfish commented:
Thanks for approving @phargogh , I added one more small fix that I just noticed. I added a test to make sure the `profile` data is applied even when there is a pre-existing metadata document. And I added some `super()` calls to make sure we are doing a roundtrip write & read for the metadata attributes that are custom classes. Otherwise they are read back in as plain `dicts`.

phargogh reviewed (approved):
No problem at all, and looks great!
